hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -7050,11 +7053,15 @@ String prepareBulkLoad(byte[] family, String srcPath, boolean copyFile)
           HStore store = getStore(family);
           try {
             if (this.rsServices != null && store.needsCompaction()) {
-              this.rsServices.getCompactionRequestor().requestCompaction(this, store,
-                "bulkload hfiles request compaction", Store.PRIORITY_USER + 1,
-                CompactionLifeCycleTracker.DUMMY, null);
-              LOG.debug("bulkload hfiles request compaction region : {}, family : {}",
-                this.getRegionInfo(), family);
+              if (!this.rsServices.getCompactionRequestor().isUnderCompaction(store)) {
```

Comment:
Will needsCompaction returns true if the store is already under compaction?

Reply:
Yes, the compaction of a store has several steps,
1. if needCompact(), add a compaction runner of the store to the queue;
2. select files to compact;
3. do compact the selected files.
If a store is added a compaction runner, but waiting in the queue, then next needCompact() is still true, and it will be add another compaction runner to queue. 
The isUnderCompaction only means whether a store has added a compaction runner to the queue.